docs/proposals/scheduling/average-replicas-assign/average-replicas-assign.md

```diff
@@ -0,0 +1,275 @@
+---
+title: Average Assign replicas strategy for Karmada
+authors:
+- ipsum-0320
+reviewers:
+- TBD
+approvers:
+- RainbowMango
+creation-date: 2024-06-28
+
+---
+
+# Average Assign replicas strategy for Karmada
+
+## Summary
+
+In Karmada's current replica scheduling policy, we have supported four replica allocation methods: replication, static weight, aggregation based on available replicas, and dynamic weight based on available replicas.
+
+<!--
+在 Karmada 当前的调度器副本数分配策略中，我们已经支持复制、静态权重、基于可用副本数的聚合、基于可用副本数的动态权重这四种副本分配方式。
+-->
+
+However, for more complex scheduling scenarios, such as ensuring that the workload replicas are evenly distributed across selected clusters after applying cross-AZ distribution constraints, Karmada currently does not support these scenarios well.
+
+<!--
+但是对于更为复杂的调度场景，例如，在跨 AZ 分布约束之后，使工作负载的副本数尽可能地平均分配在所选择的集群中。目前 Karmada 还无法很好地支持这类场景。
+-->
+
+Therefore, we plan to introduce a new replica allocation strategy called AverageReplicas for Karmada's scheduler. This strategy will support evenly distributing the target replicas among the selected clusters, and it will:
+
+1. Adhere to spread constraints.
+2. Consider the available resources in the member clusters.
+
+<!--
+因此我们计划为 Karmada 的调度器引入一种新的副本分配策略 AverageReplicas，该策略能够支持在选定的集群中均匀分配目标副本，并且该策略将：
+1.遵守传播约束。
+2.考虑工作集群中的可用资源。
+-->
+
+## Motivation
+
+By adding the AverageReplicas replica allocation strategy, Karmada can distribute the target replicas as evenly as possible across the member clusters. At the same time, this even distribution will adhere to spread constraints and consider the available resources in the member clusters.
```

Comment:
If I set it to `StaticWeight`, and it's `1:1:1:1:1`, can it achieve the `average` behavior you mentioned?

So I suggest you need to elaborate on the differences between your proposal and `StaticWeight`.

Reply:
In fact, I have already considered this, as follows.

> Therefore, we plan to introduce a new replica allocation strategy called AverageReplicas for Karmada's scheduler. This strategy will support evenly distributing the target replicas among the selected clusters, and it will:
> 1. Adhere to spread constraints.
> 2. Consider the available resources in the member clusters.

The StaticWeight does not consider spread constraints. There is some infomation about your question:

[click me](https://github.com/karmada-io/karmada/issues/4805#issuecomment-2069026112)

Reply:
How about adding that difference to the current proposal?

Reply:
+1

Reply:
> How about adding that difference to the current proposal?

yeah, i have added.

Reply:
Looking forward to your feedback on https://github.com/karmada-io/karmada/issues/4805#issuecomment-2283607217.

For now, I can see two approaches:
1. Enhance the StaticWeight 
2. Introduce a new `AverageReplicas` as proposed by this PR.

No matter which solution we are going to take, it's a great chance for us to revisit the StaticWeight feature, and make a clear behavior.